### PR TITLE
Embed python interpreter in unit-test

### PR DESCRIPTION
**Pull request by silee2**

### CODE REVIEW

File: test/backend/numpy.in.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    auto locals = py::dict("a"_a = a, "b"_a = b);
+    py::exec(R"(
+        c = a + b
+    )",
+             py::globals(),
+             locals);
+
+    auto sum = locals["c"].cast<int>();
+
+    EXPECT_TRUE(sum == a + b);
```
Comment (gaurides) [inline test/backend/numpy.in.cpp:68]: Should "a+b" go through the backend? if not, then do we need the "backend" in line 55?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -14,6 +14,36 @@
...
 # ******************************************************************************
 
+if(NGRAPH_UNIT_TEST_NUMPY_ENABLE)
+    include(ExternalProject)
+
+    ExternalProject_Add(
+        ext_pybind11
+        PREFIX ut_pybind11
+        GIT_REPOSITORY "https://github.com/pybind/pybind11.git"
+        GIT_TAG "v2.5.0"
```
Comment (gaurides) [inline test/CMakeLists.txt:24]: Is there a way to make sure the TAG here and in python/CMakeLists.txt is same?

Comment (rkimballn1) [reply test/CMakeLists.txt:24]: Yes there is, defined it in the top-level cmake file and then reference it in the two places

File: test/backend/numpy.in.cpp (hunk 1 above)
Comment (silee2) [reply test/backend/numpy.in.cpp:68]: Will remove line 55.

Comment (silee2) [reply test/backend/numpy.in.cpp:66]: Removed.

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -14,6 +14,36 @@
 # limitations under the License.
 # ******************************************************************************
 
+if(NGRAPH_UNIT_TEST_NUMPY_ENABLE)
```
[the remark this answers is not in the document]
Comment (silee2) [reply test/CMakeLists.txt]: Created cmake/external_pybind11.cmake

**Pull request merged**